sdr_airspy.c

```diff
@@ -0,0 +1,467 @@
+        }
+
+        if (AIRSPY.enable_lna_agc) {
+            status = airspy_set_lna_agc(AIRSPY.dev, 1);
+            AIRSPY_STATUS(status, "airspy_set_lna_agc failed");
+            fprintf(stderr, "LNA Gain: AGC, ");
+        } 
+        else {
+            status = airspy_set_lna_gain(AIRSPY.dev, AIRSPY.lna_gain);
+            AIRSPY_STATUS(status, "airspy_set_lna_gain failed");
+            fprintf(stderr, "LNA Gain: %i, ", AIRSPY.lna_gain);
+        }
+
+        status = airspy_set_vga_gain(AIRSPY.dev, AIRSPY.vga_gain);
+        AIRSPY_STATUS(status, "airspy_set_vga_gain failed");
+        fprintf(stderr, "VGA Gain: %i, ", AIRSPY.vga_gain);
+    }
+
+    if (AIRSPY.enable_airspy_biast) {
+        status = airspy_set_rf_bias(AIRSPY.dev, 1);
+        AIRSPY_STATUS(status, "airspy_set_rf_bias (on) failed");
+        fprintf(stderr, "Bias-t: On\n");
+    }
+    else {
+        status = airspy_set_rf_bias(AIRSPY.dev, 0);
+        AIRSPY_STATUS(status, "airspy_set_rf_bias (off) failed");
+        fprintf(stderr, "Bias-t: Off\n");
+    }
+
+    AIRSPY.converter = init_converter(INPUT_UC8,
+                                      Modes.sample_rate,
+                                      Modes.dc_filter,
+                                      &AIRSPY.converter_state);
+
+    if (!AIRSPY.converter) {
+        fprintf(stderr, "airspy: can't initialize sample converter\n");
+        airspyClose();
+        return false;
+    }
+
+    fprintf(stderr, "Airspy successfully initialized\n");
+
+    return true;
+}
+
+static struct timespec airspy_thread_cpu;
+
+int airspyCallback(airspy_transfer *transfer) {
+    int16_t *inptr = (int16_t *)transfer->samples;
+    int16_t *outptr = (int16_t *)AIRSPY.airspy_scratch;
+    size_t i_len = transfer->sample_count;
+    size_t i, i_done, o_done;
+    struct mag_buf *outbuf;
+    struct mag_buf *lastbuf;
+    uint32_t slen;
+    unsigned next_free_buffer;
+    unsigned free_bufs;
+    unsigned block_duration;
+    static int dropping = 0;
+    static uint64_t sampleCounter = 0;
+
+    // Lock the data buffer variables before accessing them
+    pthread_mutex_lock(&Modes.data_mutex);
+    if (Modes.exit) {
+        airspy_stop_rx(AIRSPY.dev);
+        return 0;
+    }
+
+    next_free_buffer = (Modes.first_free_buffer + 1) % MODES_MAG_BUFFERS;
+    outbuf = &Modes.mag_buffers[Modes.first_free_buffer];
+    lastbuf = &Modes.mag_buffers[(Modes.first_free_buffer + MODES_MAG_BUFFERS - 1) % MODES_MAG_BUFFERS];
+    free_bufs = (Modes.first_filled_buffer - next_free_buffer + MODES_MAG_BUFFERS) % MODES_MAG_BUFFERS;
+
+    // Give the demodulater what it expects
+    soxr_process(AIRSPY.resampler, inptr, i_len, &i_done, outptr, MODES_RTL_BUF_SIZE, &o_done);
+    for (i = 0; i < o_done; i++)
+        AIRSPY.airspy_bytes[i] = (int8_t)(outptr[i] >> 4) + 127;
+
+    // The resampler often produces an odd sample count, no real value in tracking how often
+    if (o_done & 1)
+        o_done -= 1;
```

Comment:
Built/installed using piaware_builder on the Pi (jessie lite), and the same for dump1090-fa using the dev-airspy build. It's been running fine until yesterday, when I decided to try airspy_adsb. It worked as far as feeding dump1090 on port 30104, but gave the same errors in the log with faup and mlat-client. I did modify dump1090-fa to use device-type none, and tried various combinations of using/removing the net-bo-port 30005 switch in attempt to get mlat working with airspy-adsb... but no joy. Having had my fill of that, I tried to switch back today and the faup/mlat-client errors remain.

It may be messy, but here is the current contents of dump1090-fa:

RECEIVER_OPTIONS="--device-type airspy --linearity-gain 21 --enable-bit-packing --enable-airspy-biast --stats --net-bo-port 30005"
NET_OPTIONS="--net --net-heartbeat 60 --net-ro-size 1000 --net-ro-interval 1 --net-ri-port 0 --net-ro-port 30002 --net-sbs-port 30003 --net-bi-port 30004,30104 --
net-bo-port 30005"
JSON_OPTIONS="--json-location-accuracy 1"

Don't think I modified anything else, but can't be absolutely sure. If the --net-bo-ipadddr supposed to be empty in the faup command line? I noticed that switch isn't available in the dev branch with dump1090.